### DISCUSSION TITLE
Copying tests from the source tree

### DISCUSSION THREAD

**Discussion by HappySeaFox** [Homebrew/discussions#4909] [Writing Formulae/Casks]
### Output of `brew config`

```shell
-
```

### Output of `brew doctor`

```shell
-
```

### Description of issue

Hi! :)

I'm developing a formula with a `test` block. The source code I build already has a couple of useful tests that can be built in the `test` block. I currently stuck on copying the tests so they can be available in the `test` block. Is there a good way to do that?

I searched across the existing formulas and found two possible ways other devs usually use:

1. `pkgshare.install` - I don't like this approach as the final package will install useless tests for the end-user.
2. Using a `resource` block - Again I don't like as it downloads files from the Internet.

Is there a better way to copy tests from the build tree?

**Comment by SMillerDev** [comment]
The best option is to follow https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula, which mentions some options for formula tests. Formula tests are intended to validate the install, so test code from upstream rarely meets the criteria for a good formula test, since those are focussed on validating implementation.

**Reply by HappySeaFox** [reply]
In this particular use-case the formula is a library and the library perfectly fits into this recommendation from the link:

_If the formula is a library, compile and run some simple code that links against it._

The tests I try to re-use do exactly what is recommended - they try to link against the installed library to make sure the installation is correct.

This is why I just want to copy-paste them from the source tree to make accessible in the test block.
